src/TraceEvent/TraceEvent.Tests/EventPipeParsing.cs

```diff
@@ -513,6 +513,60 @@ public void WellKnownDiagnosticSourceEventsHavePayloads()
             Assert.Equal(7, diagSourceEventCount);
         }
 
+        [Fact]
+        public void ExecutionContextEventsAndTimeStamping()
+        {
+            PrepareTestData();
+
+            const string eventPipeFileName = "eventpipe-dotnetcore6.0-win-x64-executioncheckpoints.nettrace";
+
+            string eventPipeFilePath = Path.Combine(UnZippedDataDir, eventPipeFileName);
+
+            var checkpoints = new Dictionary<string, double>();
+            checkpoints.Add("RuntimeInit", -15.97);
+            checkpoints.Add("RuntimeSuspend", 11.91);
+            checkpoints.Add("RuntimeResumed", 12.20);
+
+            UnicodeEncoding unicode = new UnicodeEncoding();
+            using (var source = new EventPipeEventSource(eventPipeFilePath))
+            {
+                source.Dynamic.All += (TraceEvent obj) =>
+                {
+                    ushort eventID = (ushort)obj.ID;
+
+                    // ExecutionCheckpoint.
+                    if (eventID == 300)
```

Comment:
Question on this: Do we want to update `ClrTraceEventParser` to allow this new event to be parsed using TraceEvent?  I am assuming the answer is yes, since this is the way to consume events from EventPipe.  If so, I would recommend including that update in this PR, and then you can simplify this test to just use the parser to parse the event.  Sorry, it didn't occur to me to mention this ahead of time.

Reply:
Yeah, I was about to ask that question when implementing the test, so whats the process in updating ClrTraceEventParser to include this new event currently in dotnet/runtime main branch?

Reply:
The best way to do this is to use run `TraceParserGen` against the manifest in the runtime repo.  `TraceParserGen` lives here: https://github.com/microsoft/perfview/tree/main/src/TraceParserGen.  You'll get a C# file out of this, and I would recommend just copying out the parts of the file for your event into https://github.com/microsoft/perfview/blob/main/src/TraceEvent/Parsers/ClrTraceEventParser.cs.  Theoretically, you're supposed to be able to run `TraceParserGen` and just overwrite the file, but there are customizations, and other issues that prevent this.

Once this is done, then you should just be able to consume the event by calling `source.Clr.YourEvent += delegate(YourEventData data { // do stuff here };`

Reply:
@brianrob Added parsing into ClrTraceEventParser handling both ExecutionCheckpoint emitted from clr provider (currently not emitted) and ExecutionCheckpointDCEnd emitted by clr rundown provider. Updated test to parse the clr rundown ExecutionCheckpoint events.